test/common/network/connection_impl_test.cc

```diff
@@ -573,6 +575,116 @@ TEST_P(ConnectionImplTest, BindFailureTest) {
   dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
 }
 
+class ConnectionImplBytesSentTest : public testing::TestWithParam<Address::IpVersion> {
```

Comment:
If we are using a mock transport, does this need to be parameterized on IP version?